analyzers/its/expected/Automapper/AutoMapper--net461-S2445.json

```diff
@@ -0,0 +1,17 @@
+{
+"issues":  [
+{
```

Comment:
By looking at this test case, I have some doubts about the validity of the rule on local method variables.

The code for this IT is the following:
```cs
TypeMap ResolveTypeMap(in TypePair typePair)
{
    if (_resolvedMaps.TryGetValue(typePair, out TypeMap typeMap))
    {
        return typeMap;
    }
    if (_sealed)
    {
        typeMap = _runtimeMaps.GetOrAdd(typePair);
        // if it's a dynamically created type map, we need to seal it outside GetTypeMap to handle recursion
        if (typeMap != null && typeMap.MapExpression == null)
        {
            lock (typeMap)
            {
                typeMap.Seal(this, null);
            }
        }
    }
    else
    {
        typeMap = GetTypeMap(typePair);
        _resolvedMaps.Add(typePair, typeMap);
    }
    return typeMap;
}
```

While `typeMap` is a local variable, which can mutate, it is read from `_runtimeMaps.GetOrAdd(typePair)`, which is a `private readonly LockingConcurrentDictionary<TypePair, TypeMap>`.

So basically the variable itself is not immutable, but the collection from which is read it is, and such a collection is used to have fine-graned locking based on `TypePair`.

While this all seems perfectly legitimate to me, and even necessary when you need a variable number of locks, I don't see how to easily distinguish between this scenario and the other ones.

Do we accept it as a false positive, or do we reduce the scope of the rule, to just ignore local variables?

What do you think?

Reply:
sorry I didn't reply. If it's an edge case, it's acceptable to have an FP.

Reply:
While it appears in our ITs, it seems to me enough rare, to consider it as an FP.